src/js/services/ng-scores.js

```diff
@@ -483,13 +483,11 @@ define('services/ng-scores',[
                 board[stage] = [];
             });
 
-            // Walk all scores, and put them in the corresponding round of each stage.
-            // This also performs sanity checks on the scores, marking failures.
-            // Highest scores and rankings are computed later.
-            this._rawScores.forEach(function(_score) {
-                // Create a copy of the score, such that we can add
-                // additional info
-                var s = {
+            // Create a copy of the score, such that we can add
+            // additional info (e.g. validation errors) without
+            // polluting rawScores.
```

Comment:
It may be too much for this commit, but rather than creating a copy only to be able to add meta info, I'd rather have a meta info object from which it can be fetched. Compare this to splitting tables in a database. E.g:

    {
        [_score.file]: {
            modified: false,
            error: null
        }
    }

Reply:
True, but the intention of this PR is to not change existing behaviour other than necessary for #245.

The whole project is full of 'polluting' state everywhere, which should indeed be improved. Seems that that was encouraged by angular 1.

Reply:
Not by angular per se, rather the limited skills of the author at that time (I encourage you to look up who wrote that ;))